bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpServerHandler.java

```diff
@@ -0,0 +1,471 @@
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No UDN configured for " + thing.getLabel());
+        }
+    }
+
+    private void initServer() {
+        rendererStateOptionList = new ArrayList<>();
+        upnpRenderers.forEach((key, value) -> {
+            StateOption stateOption = new StateOption(key, value.getThing().getLabel());
+            rendererStateOptionList.add(stateOption);
+        });
+        updateStateDescription(rendererChannelUID, rendererStateOptionList);
+
+        getProtocolInfo();
+
+        browse(currentEntry.getId(), "BrowseDirectChildren", "*", "0", "0", config.sortcriteria);
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handle command {} for channel {} on server {}", command, channelUID, thing.getLabel());
+
+        switch (channelUID.getId()) {
+            case UPNPRENDERER:
+                if (command instanceof StringType) {
+                    currentRendererHandler = (upnpRenderers.get(((StringType) command).toString()));
+                    if (config.filter) {
+                        // only refresh title list if filtering by renderer capabilities
+                        browse(currentEntry.getId(), "BrowseDirectChildren", "*", "0", "0", config.sortcriteria);
+                    }
+                } else if ((command instanceof RefreshType) && (currentRendererHandler != null)) {
+                    updateState(channelUID, StringType.valueOf(currentRendererHandler.getThing().getLabel()));
+                }
+                break;
+            case CURRENTID:
+                String currentId = "";
+                if (command instanceof StringType) {
+                    currentId = String.valueOf(command);
+                } else if (command instanceof RefreshType) {
+                    currentId = currentEntry.getId();
+                    updateState(channelUID, StringType.valueOf(currentId));
+                }
+                logger.debug("Setting currentId to {}", currentId);
+                if (!currentId.isEmpty()) {
+                    browse(currentId, "BrowseDirectChildren", "*", "0", "0", config.sortcriteria);
+                }
+            case BROWSE:
+                if (command instanceof StringType) {
+                    String browseTarget = command.toString();
+                    if (browseTarget != null) {
+                        if (!UP.equals(browseTarget)) {
+                            final String target = browseTarget;
+                            try {
+                                currentEntry = entries.stream().filter(entry -> target.equals(entry.getId()))
+                                        .findFirst().get();
+                            } catch (NoSuchElementException e) {
```

Comment:
It's bad practice to catch unchecked exceptions. You could use `ifPresent()` to execute the code in a lambda or `isPresent()` to check via an `if`.